proposals/public-suffix.md

```diff
@@ -0,0 +1,555 @@
+then it must also end with a shorter suffix in the ICANN section. The question
+of which one should be taken to be the domain's public suffix depends on the
+specific use case.
+
+##### Example
+
+| Domain name         | Private suffix  | ICANN suffix |
+|:-------------------:|:---------------:|:------------:|
+| foo.bar.wixsite.com | wixsite.com     | com          |
+
+#### 3. Known vs Unknown
+
+If a domain name does not match any suffix in the PSL, it is considered to have
+an unknown suffix. Depending on the use case, the public suffix could be taken
+by default to be the last domain label of the domain name, or alternatively
+the domain name could be considered invalid.
+
+**Note:** it may be more performant to allow unknown suffixes and assume a single-label
+suffix by default, because it allows the following optimisation of the lookup algorithm:
+all single-label suffixes in the PSL can be excluded from the lookup, since they do not
+need to be matched specifically.
+
+#### 4. IDN
+
+The PSL uses Unicode, and contains International Domain Name (IDN) suffixes. Depending
+on the use case, when returning the *registrable domain / eTLD+1* for a domain name,
+either Unicode or Punycode may be preferred.
+
+**Note:** the [PSL algorithm](https://github.com/publicsuffix/list/wiki/Format#formal-algorithm)
+requires Punycode for the matching logic. Therefore a requirement to convert Punycode
+back to Unicode involves extra work. It may be preferable to avoid this in
+performance-sensitive use cases.
+
+#### Use Cases
+
+#### 1. Filter Requests by Organization
+
+This API is relevant to any extension that does both of the following:
+
+* acts on web requests automatically while the user browses
+* allows the user to choose which domain names the extension should act on,
+and which should be ignored by the extension
+
+In effect, such extensions allow users to create request filtering rules that
+restrict the extension to acting on only web requests having specific domain names.
+
+When choosing domain names for the request filter, users may want to be able to choose
+all domain names for an organization without having to enter them each in individually.
+The user may want to specify request filtering rules as follows:
+
+| # | Filtering rule type | User value | Examples of domains affected |
+|:-:|---------------------|:----------:|-----------------------------:|
+| 1 | organizational base domain and all possible subdomains | myorg.co.uk | myorg.co.uk<br>subdomain1.myorg.co.uk<br>subdomain2.myorg.co.uk |
+| 2 | organization name and all possible suffixes | myorg | myorg.co.uk<br>myorg.com<br>myorg.net |
+
+To support this use case, such extensions may:
+
+* automatically calculate the set of organization names and *registrable domain / eTLD+1*s
+from a user-specified set of domain names, and propose these as filtering rules for
+the user to choose
+* prevent users from mistakenly specifying a *public suffix / eTLD* as a filtering rule,
```

Comment:
This would probably be a mistake. There are unresolved, lengthy, ancient discussions on this elsewhere but basically, public suffixes sometimes have websites.